### PR TITLE
Differentiation of REJECTED and FAILED deposit state

### PR DESCRIPTION
**Pull request by vesaakerman**
fixes EASY-1146

When applied it will

* throw `RejectedDepositException` when deposit state should be set to **REJECTED**
* give in `RejectedDepositException` the message that will be relayed to the client

#### related pull requests on github
repo                       | PR
-------------------------- | -----------------
easy-ingest-dispatcher	     |  [41]
easy-ingest-flow    	     |  [61]

@DANS-KNAW/easy

### CODE REVIEW

File: src/main/scala/nl/knaw/dans/easy/stage/package.scala [hunk 1]
```diff
@@ -18,11 +18,14 @@ package nl.knaw.dans.easy
...
 
 import nl.knaw.dans.pf.language.ddm.handlermaps.NameSpace
-import nl.knaw.dans.pf.language.emd.types.EmdConstants
 
-import scala.util.Try
+import scala.util.{Failure, Success, Try}
 
 package object stage {
+
+  // exceptions
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/stage/package.scala]: Unnecessary comment. After all, the thing is called a `RejectedDeposit`**`Exception`**.

File: src/main/scala/nl/knaw/dans/easy/stage/package.scala [hunk 2]
```diff
@@ -18,11 +18,14 @@ package nl.knaw.dans.easy
...
+  case class RejectedDepositException(msg: String = "", cause: Throwable = null) extends Exception(msg, cause)
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/stage/package.scala]: - I think we should not have a default empty message. You should always report something back to the client.
- Not sure if we would use the `Throwable` at all. From today's discussion I understood the idea was to log everything right way and not propagate all the details upstairs.

File: src/main/scala/nl/knaw/dans/easy/stage/dataset/Util.scala [hunk 3]
```diff
@@ -68,7 +69,7 @@ object Util {
   private def loadXML(fileName: String)(implicit s: Settings): Elem = {
     val metadataFile = new File(s.bagitDir, fileName)
     if (!metadataFile.exists) {
-      throw new scala.RuntimeException(s"Unable to find `$fileName` in bag.")
+      error(s"Unable to find `$fileName` in bag.")
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/stage/dataset/Util.scala:72]: Good idea to use standard functions! 👍

**Review by janvanmansum — CHANGES_REQUESTED** [review]
Almost ready for merge. Can be improve a bit; see my comments.

**Review by janvanmansum — CHANGES_REQUESTED** [review]
Almost ready for merging.

File: src/main/scala/nl/knaw/dans/easy/stage/package.scala (hunk 2 above)
Comment (vesaakerman) [reply src/main/scala/nl/knaw/dans/easy/stage/package.scala]: The idea of logging the error right away was in combination of using Either in place of Try.  Now the idea is indeed to propagate also the details upstairs, and do logging in one central place.

**Review by rvanheest — APPROVED** [review]
Looks good to me!

**Pull request merged**